### ISSUE TITLE
 [a11y] Incorrect position number of list item inside the dropdown button.

### ISSUE THREAD

**Issue by danatolie** [microsoft/fluentui#21368]
### Environment Information

- **Package version(s)**: Fluent ui 8.49.7
- **Browser and OS versions**:  Version 97.0.1072.62 (Official build) (x86_64), MacOS Version 11.6.1 (20G224)
--- Reproducible in Edge, Chrome and Safari 

### Describe the issue:

VoiceOver announces incorrect dropdown item position:
<img width="1440" alt="Screen Shot 2022-01-20 at 3 06 43 PM" src="https://user-images.githubusercontent.com/98131113/150438403-7b91996f-3e43-4703-8b72-fc1dc5f97db3.png">
 The dropdown items are:
```const options: IDropdownOption[] = [
  { key: 'test', text: 'test business' },
  { key: 'divider', text: '-', itemType: DropdownMenuItemType.Divider },
  { key: 'header', text: 'switch to', itemType: DropdownMenuItemType.Header },
  { key: 'new', text: 'new booking calendar' },
  { key: 'existing', text: 'existing booking calendar' },
];
```

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/danatolie/pen/vYeqxXd?editors=0010

#### Actual behavior:
Voiceover announces "new booking calendar not selected, switch to, group (3 of 3)"

#### Expected behavior:
Voiceover should announce "new booking calendar selected, switch to, group (2 of 3)"

There are only 3 selectable dropdown items and focus is on the second one, so it should announce (2 of 3) rather than (3 of 3)

**Comment by smhigley** [comment]
VoiceOver is treating the first option as (1 of 3), the separator as (2 of 3), and the group as (3 of 3). It looks like the two options within the group are not counted as children of the listbox at all. You can see the same behavior in the Grouped Listbox example on the APG, where each group is treated as an indexed item, but the inner options are not: https://w3c.github.io/aria-practices/examples/listbox/listbox-grouped.html.

This looks like a bug we need to file on Safari
